### PR TITLE
Video adaptation api (up to date)

### PR DESCRIPTION
**Pull request by yeshaokai**
Updated test scripts for both video inference api and video adaptation api

Restructure the code such that it's cleaner

Supports passing video_adapt = True to video_inference_superanimal to turn on video adaptation on the fly

### CODE REVIEW

File: deeplabcut/utils/auxfun_models.py [hunk 1]
```diff
@@ -157,6 +157,29 @@ def tarfilenamecutting(tarf):
         print("Pick one of the following: ", models)
 
 
+def download_mpii_weights(wd):
```
Comment (jeylau) [inline deeplabcut/utils/auxfun_models.py]: This can be removed, as weights are hosted at HF

File: deeplabcut/create_project/modelzoo.py [hunk 2]
```diff
@@ -261,7 +261,9 @@ def create_pretrained_project(
         download_huggingface_model(model, train_dir)
 
         pose_cfg = deeplabcut.auxiliaryfunctions.read_plainconfig(path_train_config)
-        pose_cfg["dataset_type"] = "imgaug"
+        # Add some multi-animal defaults
+        pose_cfg["nmsradius"] = 5.0
+        pose_cfg["sigma"] = 1
```
Comment (jeylau) [inline deeplabcut/create_project/modelzoo.py]: Revert this change: it's not needed with the line above pose_cfg["dataset_type"] = "imgaug"

File: deeplabcut/create_project/modelzoo.py [hunk 3]
```diff
@@ -296,6 +298,8 @@ def create_pretrained_project(
             "global_scale",
             "location_refinement",
             "locref_stdev",
+            "nmsradius",
+            "sigma",
```
Comment (jeylau) [inline deeplabcut/create_project/modelzoo.py]: can remove

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 4]
```diff
@@ -160,13 +165,18 @@ def after_adapt_inference(self, **kwargs):
...
         # spatial pyramid is not for adapted model
 
-        deeplabcut.video_inference_superanimal(
+        
+        superanimal_inference.video_inference(
             [self.video_path],
             self.supermodel_name,
             videotype=self.videotype,
             init_weights=adapt_weights,
+            scale_list = kwargs['scale_list'] if 'scale_list' in kwargs else [],
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]: Since you pop the key a couple of lines underneath, we could rather do:
```
scale_list = kwargs.pop('scale_list', [])
```

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py [hunk 5]
```diff
@@ -27,15 +27,16 @@
...
-
+from functools import lru_cache
 
 @PoseDatasetFactory.register("multi-animal-imgaug")
 class MAImgaugPoseDataset(BasePoseDataset):
     def __init__(self, cfg):
         super(MAImgaugPoseDataset, self).__init__(cfg)
 
-        if cfg.get("pseudo_label", ""):
+        if cfg.get("pseudo_label", "")!="":
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py]: not needed

File: deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py [hunk 6]
```diff
@@ -61,15 +62,17 @@ def train(
...
 
+    
     if modelfolder != "":
         cfg["log_dir"] = modelfolder
         cfg["project_path"] = modelfolder
         # have to overwrite this
         cfg["snapshot_prefix"] = os.path.join(modelfolder, "snapshot")
 
-    if pseudo_labels != "":
-        cfg["pseudo_label"] = pseudo_labels
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py:72]: We should keep those

File: deeplabcut/pose_estimation_tensorflow/predict_supermodel.py [hunk 7]
```diff
@@ -3,344 +3,17 @@
...
-    destfolder=None,
-    batchsize=1,
-    robust_nframes=False,
-    allow_growth=False,
-    init_weights = "",
-    customized_test_config="",
+    scale_list = [],
+    videotype = '.mp4',
+    video_adapt = False,
+    pcutoff = 0.1
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/predict_supermodel.py]: let's run black afterwards on these files; there should be no spacing on the sides of the = sign.

File: docs/ModelZoo.md [hunk 8]
```diff
@@ -1,17 +1,21 @@
 # The DeepLabCut Model Zoo! 
 
-🦒 🐈 🐕‍🦺 🐀 🐁 🦡 🦦 🐏 🐫 🐆 🦓 🐖 🐄 🐂 🦖 🐿 🦍 🦥
```
Comment (jeylau) [inline docs/ModelZoo.md]: revert

**Review by jeylau — CHANGES_REQUESTED** [review]

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 9]
```diff
@@ -2,9 +2,9 @@
 import glob
 import os
 from deeplabcut.modelzoo.utils import parse_available_supermodels
+from . import superanimal_inference
```
Comment (MMathisLab) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]: no . imports ?

**Review by MMathisLab — CHANGES_REQUESTED** [review]

**Comment by yeshaokai** [comment]
Should be ready for merging into develop?

File: deeplabcut/create_project/modelzoo.py [hunk 10]
```diff
@@ -261,7 +261,6 @@ def create_pretrained_project(
         download_huggingface_model(model, train_dir)
 
         pose_cfg = deeplabcut.auxiliaryfunctions.read_plainconfig(path_train_config)
-        pose_cfg["dataset_type"] = "imgaug"
```
Comment (jeylau) [inline deeplabcut/create_project/modelzoo.py]: We have to keep this line @yeshaokai

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 11]
```diff
@@ -142,12 +145,14 @@ def adaptation_training(self):
...
+        self.train_without_project(pseudo_label_path, **kwargs)
 
-    def after_adapt_inference(self, **kwargs):
+    def after_adapt_inference(self,
+                              apply_filter = False,
+                              **kwargs):
 
         pattern = os.path.join(
-            self.modelfolder,  f"snapshot-{self.adapt_iterations}.index"
+            self.modelfolder, f"*/snapshot-{self.adapt_iterations}.index"
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]: The path separator should not be hardcoded here

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 12]
```diff
@@ -160,12 +165,15 @@ def after_adapt_inference(self, **kwargs):
...
         # spatial pyramid is not for adapted model
 
-        deeplabcut.video_inference_superanimal(
+        scale_list = kwargs.pop('scale_list', [])        
+        superanimal_inference.video_inference(
             [self.video_path],
             self.supermodel_name,
             videotype=self.videotype,
             init_weights=adapt_weights,
+            scale_list = scale_list,
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]:
```suggestion
            scale_list=scale_list,
```

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 13]
```diff
@@ -160,12 +165,15 @@ def after_adapt_inference(self, **kwargs):
...
             customized_test_config=self.customized_pose_config,
+            apply_filter = apply_filter
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]:
```suggestion
            apply_filter=apply_filter
```

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 14]
```diff
@@ -89,32 +84,40 @@ def __init__(
...
-            videotype=self.videotype,
-            filtered=False,
-            init_weights=self.init_weights,
-            draw_skeleton=True,
-            superanimal_name=self.supermodel_name,
-            pcutoff=self.pcutoff,
-            **kwargs
-        )
+    def before_adapt_inference(self,
+                               make_video = False,
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]:
```suggestion
                               make_video=False,
```

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 15]
```diff
@@ -15,8 +15,7 @@ def __init__(
         adapt_iterations=1000,
         modelfolder="",
         customized_pose_config="",
-        pcutoff=0.1,
-        pseudo_threshold=0.1,
+        init_weights = "",
```
Comment (jeylau) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]:
```suggestion
        init_weights="",
```

File: deeplabcut/create_project/modelzoo.py (hunk 10 above)
Comment (yeshaokai) [reply deeplabcut/create_project/modelzoo.py]: Added back.

File: deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py [hunk 16]
```diff
@@ -38,10 +38,11 @@ def train(
     max_to_keep=5,
     keepdeconvweights=True,
     allow_growth=True,
-    pseudo_labels="",
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py]: This is the line to keep, not `None`

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py [hunk 17]
```diff
@@ -175,6 +179,12 @@ def build_augmentation_pipeline(self, apply_prob=0.5):
         pipeline = iaa.Sequential(random_order=False)
 
         pre_resize = cfg.get("pre_resize")
+
+        if cfg.get("traintime_resize", False):
+            # let's hard code it
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py:184]: Why? This could precisely be the value of `traintime_resize`. How is this different from `pre_resize` though?

File: docs/ModelZoo.md [hunk 18]
```diff
@@ -4,20 +4,18 @@
 
 ## 🏠 [Home page](http://modelzoo.deeplabcut.org/)
 
-Started in 2020 and expanded in 2022, the model zoo is four things:
```
Comment (jeylau) [inline docs/ModelZoo.md:7]: Again, this change should not be removed

**Comment by jeylau** [comment]
Just left some minor comments to address, and ready to merge into `develop`.

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py (hunk 17 above)
Comment (yeshaokai) [reply deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py:184]: I set this True for datasets in which images have super large resolution. To save efforts, I hardcoded it to 400 so I don't have to tune parameters and the same resizing will be applied at inference time. No much difference from pre_resize except I kept the aspect ratio

**Comment by yeshaokai** [comment]
OK I addressed the comments above

**Review by jeylau — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]
